src/Time.cpp

```diff
@@ -259,18 +284,72 @@ Time Time::fromString(const std::string& stringTime, Time::Resolution resolution
     }
 
     struct tm tm;
-    if(NULL == strptime(mainTime.c_str(), mainFormat.c_str(), &tm))
+    if (NULL == strptime(mainTime.c_str(), mainFormat.c_str(), &tm))
     {
-        throw std::runtime_error("Time::fromString failed- Time-String '" + mainTime + "' did not match the given format '" + mainFormat +"'");
+        throw std::runtime_error(
+            "Time::fromString failed: " + mainTime + "' did not match the given "
+            "format '" + mainFormat +"'"
+        );
     }
-    // " ... not set by strptime(); tells mktime() to determine 
+
+    time_t time;
+    if(tzInfo.empty())
+    {
+        tm.tm_isdst = -1;
+        time = mktime(&tm);
+    } else
+    {
+        // full offset to UTC
+        int64_t tzOffset = Time::tzInfoToSeconds(tzInfo);
+        //// now we are in utc input
+        tm.tm_sec += tzOffset;
+
+        // mktime interpretes tm as local time
+        // We have remove the local timezone's offset (at the selected date) to UTC, to really get UTC
+        tm.tm_isdst = -1;
+        time = mktime(&tm);
+
+        // Identify offset to UTC in the current timezone at 'time'
+        int64_t localTimezoneOffset = Time::getTimezoneOffset(time);
+        // now we move to the local timezone
+        time -= localTimezoneOffset;
```

Comment:
I'm going to assume that the documentation you wrote for fromString is what you actually intend to do, which is

~~~
        * \returns a Time object representing UTC if the input string is
        *           suffixed by the offset to UTC '\%z', e.g. +0230 for two hours
        *           and 30 minutes (east of GMT), and localtime otherwise
~~~

If the intention is indeed to return a base::Time in UTC, then the only thing needed is to apply the offset contained in the string. getTimezoneOffset is not needed.